openshift/openshift/src/main/java/org/arquillian/cube/openshift/impl/resources/MethodTemplateProcessor.java

```diff
@@ -0,0 +1,127 @@
+package org.arquillian.cube.openshift.impl.resources;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import org.arquillian.cube.openshift.api.Template;
+import org.arquillian.cube.openshift.api.model.OpenShiftResource;
+import org.arquillian.cube.openshift.impl.adapter.OpenShiftAdapter;
+import org.arquillian.cube.openshift.impl.client.CubeOpenShiftConfiguration;
+import org.arquillian.cube.openshift.impl.utils.ParamValue;
+import org.arquillian.cube.openshift.impl.utils.StringResolver;
+import org.arquillian.cube.openshift.impl.utils.Strings;
+import org.arquillian.cube.openshift.impl.utils.TemplateUtils;
+import org.jboss.arquillian.test.spi.TestClass;
+
+import static org.arquillian.cube.openshift.impl.resources.OpenShiftResourceFactory.additionalCleanup;
+import static org.arquillian.cube.openshift.impl.resources.OpenShiftResourceFactory.createResourceKey;
+import static org.arquillian.cube.openshift.impl.resources.OpenShiftResourceFactory.deleteTemplates;
+
+public class MethodTemplateProcessor implements TemplateProcessor {
+
+    private static final Logger log = Logger.getLogger(MethodTemplateProcessor.class.getName());
+
+    private final OpenShiftAdapter openShiftAdapter;
+    private final CubeOpenShiftConfiguration configuration;
+    private final TestClass testClass;
+    private final Method testMethod;
+
+    private List<Template> templates = Collections.emptyList();
+
+    public MethodTemplateProcessor(OpenShiftAdapter openShiftAdapter,
+        CubeOpenShiftConfiguration configuration, TestClass testClass, Method testMethod) {
+        this.openShiftAdapter = openShiftAdapter;
+        this.configuration = configuration;
+        this.testClass = testClass;
+        this.testMethod = testMethod;
+    }
+
+    public List<Template> getTemplates() {
+        return templates;
+    }
+
+    /**
+     * Instantiates the templates specified by @Template within @Templates
+     */
+    public List<List<? extends OpenShiftResource>> processTemplateResources() {
+        List<? extends OpenShiftResource> resources;
+        final List<List<? extends OpenShiftResource>> RESOURCES = new ArrayList<>();
```

Comment:
I know this code comes from old impl. but can we name this variable in lower case to not confuse as a constant?